addons/project/views/project_sharing_project_task_views.xml

```diff
@@ -158,7 +158,7 @@
                             <field name="allow_milestones" invisible="1"/>
                             <field name="milestone_id"
                                 placeholder="e.g. Product Launch"
-                                context="{'default_project_id': project_id}"
+                                context="{'default_project_id': project_id, 'is_sharing_milestone_field': True}"
```

Comment:
IMO, it is not really needed since there is an option to avoid editing and opening milestone_id field.